File: backend/src/main/java/com/smartroute/smartroute1/entity/StravaActivity.java
package com.smartroute.smartroute1.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class StravaActivity {
    @Id
    private Long id;

    private String name;
    private float distance;
    private int movingTime;
    private int elapsedTime;
    private float totalElevationGain;
    private String type;
    private String sportType;
    private String startDate;
    private String startDateLocal;
    private float averageSpeed;
    private float maxSpeed;
    private Float averageWatts;
    private Float averageHeartrate;
    private Float maxHeartrate;
    private Float kilojoules;
    private Integer sufferScore;
    @Column(columnDefinition = "TEXT")
    private String summaryPolyline;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    private StravaAccount stravaAccount;
}

File: frontend/src/app/dtos/StravaActivity.ts
export interface DetailedStravaActivity {
  id: number;
  name: string;
  distance: number;
  movingTime: number;
  elapsedTime: number;
  totalElevationGain: number;
  type: string;
  sportType: string;
  startDate: string;
  startDateLocal: string;
  averageSpeed: number;
  maxSpeed: number;
  averageHeartrate: number | null;
  maxHeartrate: number | null;
  averageWatts: number | null;
  kilojoules: number | null;
  summaryPolyline: string|null;
}

export interface StravaActivity {
  id: number;
  name: string;
  distance: number;
  movingTime: number;
  totalElevationGain: number;
  sportType: string;
  startDateLocal: string;
  averageSpeed: number;
  averageHeartrate: number | null;
}
